File: Leetcode Daily Challenge/013-July-2024/009-average-waiting-time.java
// https://leetcode.com/problems/average-waiting-time/



class Solution {
    // TC : O(N)
    // SC : O(1)
    public double averageWaitingTime(int[][] customers) {
        int n = customers.length;

        double totalWaitTime = 0;
        int currTime         = 0;

        for(int[] vec : customers) {
            int arrivalTime = vec[0];
            int cookTime    = vec[1];

            if(currTime < arrivalTime) {
                currTime = arrivalTime;
            }

            int waitTime   = currTime + cookTime - arrivalTime;
            totalWaitTime += waitTime;
            currTime      += cookTime;
        }

        return totalWaitTime/n;
    }
}

File: Leetcode Daily Challenge/013-July-2024/009-average-waiting-time.py
# https://leetcode.com/problems/average-waiting-time/



class Solution:
    def averageWaitingTime(self, customers):
        """
        :type customers: List[List[int]]
        :rtype: float
        """
        n = len(customers)

        totalWaitTime = 0
        currTime = 0

        for vec in customers:
            arrivalTime = vec[0]
            cookTime = vec[1]

            if currTime < arrivalTime:
                currTime = arrivalTime

            waitTime = currTime + cookTime - arrivalTime
            totalWaitTime += waitTime
            currTime += cookTime

        return totalWaitTime / float(n)